factom_core/block_elements/entry_commit.py

```diff
@@ -11,16 +11,27 @@ class EntryCommit:
     entry_hash: bytes
     ec_spent: int
     ec_public_key: bytes
-    signature: bytes
+    _signature: bytes = None
 
     def __post_init__(self):
         # TODO: assert they're all here
         pass
 
+    @property
+    def signature(self):
+        return self._signature
+
+    @signature.setter
+    def signature(self, signature: bytes):
+        self._signature = signature
+
     def marshal(self):
         """Marshals the EntryCommit according to the byte-level representation shown at
         https://github.com/FactomProject/FactomDocs/blob/master/factomDataStructureDetails.md#entry-commit
         """
+        # Fail if signature is not set.
+        assert self._signature is not None
```

Comment:
change this to just `self.signature`

Also, instead of repeating the marshalling code here and in the marshal for signature. Have this function call marshal for signature and then append the signature to the end of that. Like I do in Factoid Transaction here: https://github.com/sambarnes/factom-core/blob/master/factom_core/block_elements/factoid_transaction.py#L38